### PR TITLE
Map cognito:groups to Quarkus roles

### LINKED ISSUE #28544: Map Lambda "cognito:groups" to Quarkus roles
### Discussed in https://github.com/quarkusio/quarkus/discussions/28080

<div type='discussions-op-text'>

<sup>Originally posted by **vladaman** September 20, 2022</sup>
We have integrated AWS API Gateway HTTP API + Cognito and JWT. When we include `Bearer` keyword in Authorization header we get 401 error from Quarkus.

Returns 401 error
```bash
curl -X 'GET' \
  'https://api.mysite.com/0.5/app-user/heyho' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJraWQiOiJ1aVcwcxxxxx'
```
Response:
```
< HTTP/2 401 
< date: Tue, 20 Sep 2022 08:03:32 GMT
< content-length: 0
< www-authenticate: Bearer
< apigw-requestid: Yv8JOgAWhAcEMpQ=
```

Following call succeeds (_note, Bearer keyword is not included_) but JsonWebToken is null:
```bash
curl -X 'GET' \
  'https://api.mysite.com/0.5/app-user/heyho' \
  -H 'accept: application/json' \
  -H 'Authorization: eyJraWQiOiJ1aVcwcxxxxx'
```

application.properties:
```
quarkus.lambda-http.enable-security=true
```

How do I configure Quarkus to accept Authorization header via AWS API Gateway if it includes "Bearer"?</div>

### PR DESCRIPTION
**Pull request by patriot1burke**
Fix #28544

### CODE REVIEW

**Review by geoand — COMMENTED** [review]
Seems reasonable to me, but this is more of @sberyozkin's wheelhouse, so I'll let him comment

**Comment by sberyozkin** [comment]
@geoand Thanks, I also agree it makes sense.

@patriot1burke Nice, having just an MP JWT API dependency is light and allowing users to apply JWT with amazon-lambda the same way they handle it in `quarkus-oidc`, `quarkus-smallrye-jwt` is a good idea IMHO.
I'd suggest to introduce `amazon-lamda-common` in the follow up PR to avoid duplicating, as it would likely involve moving some unrelated code if it were done in this PR, so please consider doing it a bit later.
I have one comment in mind which I'd like to discuss before approving...

**Comment by patriot1burke** [comment]
> @geoand Thanks, I also agree it makes sense.
> 
> @patriot1burke Nice, having just an MP JWT API dependency is light and allowing users to apply JWT with amazon-lambda the same way they handle it in `quarkus-oidc`, `quarkus-smallrye-jwt` is a good idea IMHO. I'd suggest to introduce `amazon-lamda-common` in the follow up PR to avoid duplicating, as it would likely involve moving some unrelated code if it were done in this PR, so please consider doing it a bit later. I have one comment in mind which I'd like to discuss before approving...

Don't want a common code base between REST/HTTP.  The differences are subtle, but they exist.

File: extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class LambdaHttpConfig {
+
+    /**
+     * If true, runtime will search Cognito JWT claims for "cognito:groups"
+     * and add them as Quarkus security roles.
+     *
+     * True by default
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean enableCognitoGroups;
```
[the remark this answers is not in the document]
Comment (patriot1burke) [reply extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpConfig.java:17]: I can add that.  Actually, that's a lot better as I couldn't find good documentation on how Cognito is transmitted with REST event and even saw some examples of custom role mappings.

**Comment by patriot1burke** [comment]
@sberyozkin Ok, role claim config is in.  I think this is good now?

**Comment by sberyozkin** [comment]
@patriot1burke Thanks, good idea to add an option to customize the role values pattern

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**